src/NUnitFramework/nunitlite/TextRunner.cs

```diff
@@ -318,21 +318,21 @@ public void ReportResults(ITestResult result)
             _textUI.DisplaySummaryReport(Summary);
         }
 
-        private int ExploreTests()
+        private int ExploreTests(ITestFilter filter)
         {
 #if !PORTABLE
             ITest testNode = _runner.LoadedTest;
```

Comment:
@CharliePoole  I have updated the runner to now have an Explore(filter) function. I have gone with changing the builder to rebuild a new test list based on a filter. I tried a number of other solutions to the problem but this seemed to be the least intrusive.

The `LoadedTest` suite `Test` list being readonly was the hardest challenge. Attempting to work around it by creating new lists was getting very cumbersome. I even attempted to remove tests based on the filter. The secondary issue I was having was attempting to clone the `LoadedTest` structure. If I didn't them the changes kept changing the `LoadedTest` internal list of tests, which was undesirable. 

Hence I went for building a new test suite based on the filter.

Let me know what you think.